### ISSUE TITLE
Protected entries and global password

### ISSUE THREAD

**Issue by AlGolUA** [sabrogden/Ditto#325]
That's an idea of using Ditto as password manager. It should provide an option from its menu to define some global password. Also it should be possible to mark some entries as protected. Content of such entries should be encypted by the global password, it should not be encrypted during search, but it's title is included. Content is decrypted just at the moment of paste and possible just if a correct global password is defined.

**Comment by AlGolUA** [comment]
Global password change procedure should be also included - all protected entries content should be decoded with old and incoded with new global password.

**Comment by garoto** [comment]
> That's an idea of using Ditto as password manager.

Against. Ditto is a simple clipboard manager, take care of your passwords `management` yourself.

**Comment by Suprr** [comment]
++ Support for this feature

**Comment by matrixmainframe** [comment]
I also support this feature request

**Comment by sjvudp** [comment]
> That's an idea of using Ditto as password manager.

There are password managers like KeePass. Passwords are quite sensitive, and shouldn't be handled by Ditto IMHO.
However I could see a benefit providing a global startup password used to encrypt the database (probably meaning: Every entry).

**Comment by hueldoeu** [comment]
i need this now.

**Comment by Chealer** [comment]
@AlGolUA: Tickets should be formulated in functional terms. In this case, what you are requesting is protected entries, not a global password. A global password would be one of multiple ways to authenticate so that entries can be decrypted.

So, please retitle this to reflect what you need only. Which brings us to your needs: [CopyQ features password protection](https://copyq.readthedocs.io/en/latest/password-protection.html#encrypt), but that does not make it a password manager. Are you only asking for protected entries here as the title suggests, or are you actually asking Ditto to become a password manager?

<details>
  <summary>License</summary>
This comment and any other from myself on this ticket is offered under the terms of CC0 1.0.
</details>

**Comment by hueldoeu** [comment]
i am sorry for writing this comment. i know that i am not the one spoken to,

i can't remember why i needed a password for my over 20 gb big database of ditto.

it has so much/many data that it became slowly for searching.

the data is quite sensitiv. it's not good if someone opens ditto from infotray/systray and sees all the stuff stored there.

ditto of course isn't a password manager. but i still wan't to protect the content with a tiny window to enter my password.

<img width="507" height="917" alt="Image" src="https://github.com/user-attachments/assets/875aed9f-04da-4e95-9141-eb0dfcc7a9a9" />

**Comment by Chealer** [comment]
@hueldoeu: If I understand correctly, you sometimes leave a PC connected to your session unattended, and with the session unlocked?

**Comment by hueldoeu** [comment]
> [@hueldoeu](https://github.com/hueldoeu): If I understand correctly, you sometimes leave a PC connected to your session unattended, and with the session unlocked?

never sometimes but always always

**Comment by AlGolUA** [comment]
Let me explain the idea by own example. What do I do now to use Ditto as password storage:
1) Enter password and copy it, so that it appears in Ditto
2) Find the entry in Ditto and go to its properties
3) Change its title to identify the entry and find it easy (for example "Office RDP password") and mark the entry as never deletable.
That's all. And than, when I need to enter the password I start searching "Office" entry and voila - I just need to press Enter to place the password in currently selected place. 
Easy and fine. But what is bad - the content is stored in Ditto database and is not encrypted there.
So, would be nice to encript similar entries (such entries should have some special flag). And their contents should not be used during searching - their titles are used and that's enough.
How to encrypt contents of such entries - is your decision, but if it will depend on some global password then it should be possible to change it with re-enconing of stored entries.

**Comment by Chealer** [comment]
Thanks @AlGolUA 

You are right that Ditto can already serve as a bare-bones password manager. It is likely that it upgrades to a very basic password manager if issue #986 is solved.

However, this request is very reminiscent of Zawinski's Law of Software Envelopment, and I fail to understand why you use Ditto for this purpose. Please explain what you find wrong in existing password managers―for example KeePassXC. Would you use protected entries for something other than passwords?

**Comment by AlGolUA** [comment]
Password, of course, is not the only example of data that deserve to be encrypted. And I'm strictly against full encryption of db since it can have a dramatic impact on search functionality. My proposal is something middle between full encryption and its absence. User should decide himself what should be encrypted (and its consent, but not title, should be skipped during search).

**Comment by Chealer** [comment]
> Password, of course, is not the only example of data that deserve to be encrypted.

Of course, but what could be other examples of entries that you would store encrypted in Ditto?

> And I'm strictly against full encryption of db since it can have a dramatic impact on search functionality.

Is the impact you are referring to more than extra computations (CPU usage) when searching?
